Bootstrapping with a custom extension in a private maven repo

**Describe the bug**

The bootstrap code does not actually know all the ways to find the settings.xml file and does not seem to pay attention to any mirrors or authentication configured in the settings.xml when it is found.

**Expected behavior**

Bootstraping should work when custom extensions are in a private maven repository

**Actual behavior**

1. Doesn't work when the settings.xml is in one of the assumed places but has a `mirrorOf` that uses authentication.
2. Doesn't work when the settings.xml location is defined by `MVN_SETTINGS` environment variable

**To Reproduce**

Scenario A - Wherein the `settings.xml` is in an expected location, e.g. `~/.m2/settings.xml`

Steps to reproduce the behavior:
1. Ensure that the `settings.xml` has been configured to use the private repo for all resolution and that auth for the server has been provided (bonus marks if the password is encrypted using the maven secret key in the `settings-security.xml` side file):
    ```
      <mirrors>
        <mirror>
          <id>your-id</id>
          <mirrorOf>*</mirrorOf>
          <url>https://your-repo-url</url>
        </mirror>
      </mirrors>
      ...
      <servers>
        <server>
          <id>your-id</id>
          <username>...</username>
          <password>...</password>
        </server>
        ...
      </servers>
    ```
2. Build any project that uses a custom extension resolved from the maven repository and with a `@QuarkusTest`. This build will probably work because Maven will resolve things for us. (IOW if you have *not* used the recommendation of `securtity-settings.xml` encryption it should work, but if you have followed the recommendations of the Apache Maven PMC and encrypted then it will not)
3. Delete the custom extension from the local repository.
4. Run the test from your IDE... behold the majestic failure stack trace:
```
java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for **REDACTED**::jar:**REDACTED**

	at io.quarkus.test.junit.QuarkusTestExtension.beforeEach(QuarkusTestExtension.java:264)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachCallbacks$1(TestMethodTestDescriptor.java:161)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$5(TestMethodTestDescriptor.java:197)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeMethodsOrCallbacksUntilExceptionOccurs(TestMethodTestDescriptor.java:197)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeEachCallbacks(TestMethodTestDescriptor.java:160)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for **REDACTED**::jar:**REDACTED**
	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:301)
	at io.quarkus.bootstrap.app.QuarkusBootstrap.bootstrap(QuarkusBootstrap.java:140)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:129)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:278)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:301)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	... 29 more
Caused by: io.quarkus.bootstrap.resolver.AppModelResolverException: Failed to resolve dependencies for **REDACTED**:jar:**REDACTED**
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveManagedDependencies(MavenArtifactResolver.java:250)
	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel(BootstrapAppModelResolver.java:177)
	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel(BootstrapAppModelResolver.java:144)
	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:287)
	... 39 more
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at **REDACTED**:jar:**REDACTED**
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:353)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveManagedDependencies(MavenArtifactResolver.java:246)
	... 42 more
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at **REDACTED**:jar:**REDACTED**
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:291)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:309)
	... 43 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for **REDACTED**:jar:**REDACTED**
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:255)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.resolveCachedArtifactDescriptor(DefaultDependencyCollector.java:541)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.getArtifactDescriptorResult(DefaultDependencyCollector.java:524)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:412)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:365)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:352)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:254)
	... 44 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact **REDACTED**:pom:**REDACTED** from/to cloudbees-internal (https://**REDACTED**): Authentication failed for https://**REDACTED**.pom 401 Unauthorized
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:424)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:207)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:240)
	... 51 more
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact **REDACTED**:pom:**REDACTED** from/to **REDACTED** (https://**REDACTED**): Authentication failed for https://**REDACTED**.pom 401 Unauthorized
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:52)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:369)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:75)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:644)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:262)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:499)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:401)
	... 54 more
Caused by: org.apache.maven.wagon.authorization.AuthorizationException: Authentication failed for https://**REDACTED**.pom 401 Unauthorized
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.fillInputData(AbstractHttpClientWagon.java:1180)
	at org.apache.maven.wagon.shared.http.AbstractHttpClientWagon.fillInputData(AbstractHttpClientWagon.java:1138)
	at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:126)
	at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)
	at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
	at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run(WagonTransporter.java:567)
	at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:435)
	at org.eclipse.aether.transport.wagon.WagonTransporter.get(WagonTransporter.java:412)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask(BasicRepositoryConnector.java:457)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:364)
	... 59 more
``` 

Scenario B - Wherein your build server is creating a `settings.xml` specifically for your build with it's own custom local repository (as recommended to prevent concurrency issues from multiple builds accessing the same local repository at the same time.

1. Move your `~/.m2` directory somewhere else, e.g. `~/.m2-alt`
2. Ensure that the `settings.xml` has been configured to use the private repo for all resolution and that auth for the server has been provided (bonus marks if the password is encrypted using the maven secret key in the `settings-security.xml` side file):
    ```
      <localRepository>${user.home}/.m2-alt/repository</localRepository>
      <mirrors>
        <mirror>
          <id>your-id</id>
          <mirrorOf>*</mirrorOf>
          <url>https://your-repo-url</url>
        </mirror>
      </mirrors>
      ...
      <servers>
        <server>
          <id>your-id</id>
          <username>...</username>
          <password>...</password>
        </server>
        ...
      </servers>
    ```
3. `export MAVEN_CONFIG="--settings ~/.m2-alt/settings.xml"`
4. Build any project that uses a custom extension resolved from the maven repository and watch it fail because there is no `settings.xml` in  the locations that Quarkus looks for (Note this search result: https://github.com/quarkusio/quarkus/search?q=MAVEN_CONFIG&unscoped_q=MAVEN_CONFIG only finds the `mvnw` wrapper script) and hence it will assume default settings and trying to pull everything from central put them incorrectly into `~/.m2/repository`

**Configuration**
N/A

**Screenshots**
N/A

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 1.8
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.4.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3

**Additional context**
https://github.com/quarkusio/quarkus/issues/8121#issuecomment-627545603

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java
@@ -423,14 +423,17 @@ public class BootstrapMavenContext {
         }
         final Artifact projectArtifact = new DefaultArtifact(ModelUtils.getGroupId(model), model.getArtifactId(), "", "pom",
                 ModelUtils.getVersion(model));
+        final List<RemoteRepository> rawRepos;
         try {
-            return getRepositorySystem().readArtifactDescriptor(getRepositorySystemSession(), new ArtifactDescriptorRequest()
-                    .setArtifact(projectArtifact)
-                    .setRepositories(repos))
+            rawRepos = getRepositorySystem()
+                    .readArtifactDescriptor(getRepositorySystemSession(), new ArtifactDescriptorRequest()
+                            .setArtifact(projectArtifact)
+                            .setRepositories(repos))
                     .getRepositories();
         } catch (ArtifactDescriptorException e) {
             throw new AppModelResolverException("Failed to read artifact descriptor for " + projectArtifact, e);
         }
+        return getRepositorySystem().newResolutionRepositories(getRepositorySystemSession(), rawRepos);
     }
 
     public List<org.apache.maven.model.Profile> getActiveSettingsProfiles()

--- a/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/maven/test/BootstrapMavenContextTestBase.java
+++ b/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/maven/test/BootstrapMavenContextTestBase.java
@@ -55,8 +55,7 @@ public class BootstrapMavenContextTestBase {
 
     protected BootstrapMavenContext bootstrapMavenContextWithSettings(String configDirOnCp) throws Exception {
 
-        final BootstrapMavenContextConfig<?> config = BootstrapMavenContext.config()
-                .setWorkspaceDiscovery(false);
+        final BootstrapMavenContextConfig<?> config = initBootstrapMavenContextConfig();
 
         final Path projectLocation = getProjectLocation(configDirOnCp);
         final Path projectSettingsXml = projectLocation.resolve("settings.xml");
@@ -64,6 +63,10 @@ public class BootstrapMavenContextTestBase {
             config.setUserSettings(projectSettingsXml.toFile());
         }
         return new BootstrapMavenContext(config);
+    }
+
+    protected BootstrapMavenContextConfig<?> initBootstrapMavenContextConfig() throws Exception {
+        return BootstrapMavenContext.config().setWorkspaceDiscovery(false);
     }
 
     protected Path getProjectLocation(String projectOnCp) throws URISyntaxException {

--- a/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/maven/test/PomReposMirroredTest.java
+++ b/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/resolver/maven/test/PomReposMirroredTest.java
@@ -1,0 +1,57 @@
+package io.quarkus.bootstrap.resolver.maven.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext;
+import io.quarkus.bootstrap.resolver.maven.BootstrapMavenContextConfig;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.junit.jupiter.api.Test;
+
+public class PomReposMirroredTest extends BootstrapMavenContextTestBase {
+
+    private static final String projectPath = "custom-settings/pom-repos-mirrored";
+
+    private static final Map<String, String> MIRRORED;
+    static {
+        final Map<String, String> tmp = new HashMap<>();
+        tmp.put("jboss-public-repository", "https://repository.jboss.org/nexus/content/repositories/releases/");
+        tmp.put("custom", "https://custom-repo.org/repo");
+        tmp.put("central", "https://repo.maven.apache.org/maven2");
+        MIRRORED = tmp;
+    }
+
+    protected BootstrapMavenContextConfig<?> initBootstrapMavenContextConfig() throws Exception {
+        final Path projectLocation = getProjectLocation(projectPath);
+        return BootstrapMavenContext.config()
+                .setWorkspaceDiscovery(true)
+                .setCurrentProject(projectLocation.toString());
+    }
+
+    @Test
+    public void basicPomRepos() throws Exception {
+        final BootstrapMavenContext mvn = bootstrapMavenContextWithSettings("custom-settings/pom-repos-mirrored");
+
+        final List<RemoteRepository> repos = mvn.getRemoteRepositories();
+        assertEquals(1, repos.size());
+
+        RemoteRepository repo = repos.get(0);
+        assertEquals("private-repo", repo.getId());
+        assertNull(repo.getProxy());
+        assertNotNull(repo.getAuthentication());
+
+        List<RemoteRepository> mirroredRepos = repo.getMirroredRepositories();
+        assertNotNull(mirroredRepos);
+        assertEquals(MIRRORED.size(), mirroredRepos.size());
+        for (RemoteRepository r : mirroredRepos) {
+            assertTrue(MIRRORED.containsKey(r.getId()));
+            assertEquals(MIRRORED.get(r.getId()), r.getUrl());
+        }
+    }
+}

--- a/independent-projects/bootstrap/core/src/test/resources/custom-settings/pom-repos-mirrored/pom.xml
+++ b/independent-projects/bootstrap/core/src/test/resources/custom-settings/pom-repos-mirrored/pom.xml
@@ -1,0 +1,14 @@
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.acme.quarkus.sample</groupId>
+  <artifactId>my-quarkus-project</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <repositories>
+    <repository>
+      <id>custom</id>
+      <url>https://custom-repo.org/repo</url>
+    </repository>
+  </repositories>
+</project>

--- a/independent-projects/bootstrap/core/src/test/resources/custom-settings/pom-repos-mirrored/settings.xml
+++ b/independent-projects/bootstrap/core/src/test/resources/custom-settings/pom-repos-mirrored/settings.xml
@@ -1,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.1.0 http://maven.apache.org/xsd/settings-1.1.0.xsd" xmlns="http://maven.apache.org/SETTINGS/1.1.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <servers>
+    <server>
+      <id>private-repo</id>
+      <username>reader</username>
+      <password>xxxxx</password>
+    </server>
+  </servers>
+  <mirrors>
+    <mirror>
+      <mirrorOf>*</mirrorOf>
+      <url>https://private.net/repo</url>
+      <id>private-repo</id>
+    </mirror>
+  </mirrors>
+  <profiles>
+    <profile>
+      <repositories>
+        <repository>
+          <id>jboss-public-repository</id>
+          <url>https://repository.jboss.org/nexus/content/repositories/releases/</url>
+        </repository>
+      </repositories>
+      <id>foo-profile</id>
+    </profile>
+  </profiles>
+  <activeProfiles>
+    <activeProfile>foo-profile</activeProfile>
+  </activeProfiles>
+</settings>

